lib/utils/wm_api.dart

```diff
@@ -80,12 +80,16 @@ class WmAPI {
   }
 
   void openApp(String packageName) {
+    final application = getApp(packageName);
+    if (!application.canBeOpened) {
+      throw 'The app couldn not be opened';
```

Comment:
This function shouldn't be called if `canBeOpened` is false. We should handle it in a try {} catch {} block as well

Reply:
i think its better to handle stuff right here because openApp could be called from multiple sources and so adding the logic here would mean having a centralized logic